src/varity/chain.clj

```diff
@@ -122,3 +133,18 @@
   [chr pos chain-idx]
   (->> (chain-idx (normalize-chr chr))
        (keep (partial in-block? pos))))
+
+(defn search-overlap-chains
+  "Calculats a list of chains that overlap the given interval."
+  [chr start end chain-idx]
+  (->> (chain-idx (normalize-chr chr))
+       (filter #(and (<= (get-in % [:header :t-start]) start)
+                     (<= end (get-in % [:header :t-end]))))))
+
+(defn search-overlap-blocks
+  "Calculates a list of blocks that overlap the given interval."
+  [start end ^Sorted blocks-idx]
+  (->> (subseq blocks-idx <= end)
+       (take-while (fn [[_ block]]
+                     (<= start (:t-end block))))
```

Comment:
Should be `(drop-while (fn [[_ block]] (<= (:t-end block) start)))`? (I'm not so confident if `=` should really be included, though 😅)

And if it's correct, that implies a few more tests are necessary to capture the difference between the current implementation and the one I suggested above. I think it would be nice to have a couple of dedicated test cases for each `varity.chain/search-*` fn.

Reply:
Thank you.My implementation is wrong.
`(drop-while (fn [[_ block]] (< (:t-end blcok)  start))`